src/shell/vsf_tools.php

```diff
@@ -29,31 +29,34 @@ public function run()
             $action = '';
         }
 
-        /**
-         * TODO add option to have full reindex per store
-         */
+        $storeId = empty($this->getArg('store')) ? null : (int) $this->getArg('store');
+
         switch ($action) {
             case 'full_reindex':
                 $type = $this->getArg('type');
-                $storeId = (int)$this->getArg('store');
-
-                if (0 === $storeId) {
-                    echo "Please provide store to reindex \n\n";
-                    echo $this->usageHelp();
 
-                    return;
+                if ($storeId === null) {
+                    /** @var Divante_VueStorefrontIndexer_Model_Config_Generalsettings $settings */
+                    $settings = Mage::getSingleton('vsf_indexer/config_generalsettings');
+                    $storeIds = $settings->getStoresToIndex();
+                } else {
+                    $storeIds = array($storeId);
```

Comment:
Maybe we should alsow check if given store is allowed to reindex (set in configuration) and show message if not ?

`if (in_array($storeId, settings->getStoresToIndex()))` or smth like that

Reply:
I added it like this:
```php
/** @var Divante_VueStorefrontIndexer_Model_Config_Generalsettings $settings */
$settings = Mage::getSingleton('vsf_indexer/config_generalsettings');

if ($storeId === null) {
    $storeIds = $settings->getStoresToIndex();
} else {
    $storeIds = ($settings->canReindexStore($storeId)) ? array($storeId) : array();
}
```

Because `$settings->getStoresToIndex()` already checks for only selected stores with `canReindexStore($storeId)`. And then I used the same method in the `else` clause.